virtualbox/provider.go

```diff
@@ -23,8 +29,41 @@ func init() {
 // Provider returns a resource provider for virtualbox.
 func Provider() *schema.Provider {
 	return &schema.Provider{
+		// Provider configuration
+		Schema: map[string]*schema.Schema{
+			"delay": &schema.Schema{
```

Comment:
The name of the variable is a bit broad. Perhaps ready_delay and ready_timeout? It would also be useful to accept a string, and using a validator to ensure it is of time.Duration. This will avoid the ambiguity of what are the units.